lib/puppet/util.rb

```diff
@@ -25,6 +25,67 @@ module Util
 
   extend Puppet::Util::SymbolicFileMode
 
+  public
+  def self.get_environment_variable(name, mode = nil)
+    if mode.nil?
+      if Puppet.features.microsoft_windows?
+        mode = :windows
+      else
+        mode = :posix
+      end
+    end
+
+    if mode == :windows
+      Puppet::Util::Windows::Process.get_environment_strings.each do |key, value |
+        if name.casecmp(key) == 0 then
+          return value
+        end
+      end
+      return nil
+    else
+      ENV[name]
+    end
+  end
+  # TODO use alias_method instead
+  def self.env(name,mode = nil)
```

Comment:
I realize you asked me about this the other day, but the ask seems clearer in the code to me now.  I think that we should use a hash-like structure, similar to `ENV` to allow getting / setting through the same interface *or* we use a `get` / `set` dual.